boring/Cargo.toml

```diff
@@ -16,10 +16,12 @@ features = ["rpk", "pq-experimental"]
 rustdoc-args = ["--cfg", "docsrs"]
 
 [features]
-# Use a FIPS-validated version of boringssl.
+# Use a FIPS-validated version of boringssl. Moreover, only support the NIST
+# curves for the key exchange by default.
 fips = ["boring-sys/fips"]
```

Comment:
Question for reviewer: "fips" and "rpk" are meant to be mutually exclusive. What about "fips" and "pq-experimental"?

Reply:
iirc according to @bwesterb it should be compatible

Reply:
Hmm, I ask because I wonder if there is a technical reason for the code below that would also apply to the PQ patch (from boring-sys/build.rs):
```rust
    #[cfg(all(feature = "fips", feature = "rpk"))]
    compile_error!("`fips` and `rpk` features are mutually exclusive");
```

Reply:
After discussing with Rushil offline, it turns out that "fips" is mutually exclusive with any feature that depends on a newer version of boringSSL or the patches.

Reply:
@bwesterb My understanding was that  the "fips" feature implies we just use the latest FIPS validated commit from the GH mirror and don't mess with it. Do we apply the PQ patch on top of that?

Reply:
Based on https://github.com/cloudflare/boring/blob/master/boring-sys/build.rs#L120-L123, it looks to me like `boringssl-fips` is used if "fips" is set and "fips-link-precompiled" is not set. Otherwise, `boringssl` is used.

The PQ patch applies to `boringssl`, but likely not `boringssl-fips` since this version of boringSSL seems to predate Kyber.

So "pq-experimental,fips-link-precompiled" should build, but "pq-experimental,fips" probably doesn't.